File: src/LCS02.WqXACV.1000226/solution.ts
function halfQuestions (questions: number[]): number {
    const N = questions.length / 2;
    const map = new Map<number, number>();
    for (const num of questions) {
        if (!map.has(num)) {
            map.set(num, 1);
        } else {
            map.set(num, map.get(num) + 1);
        }
    }
    const entries = [...map.entries(), ].sort((a, b) => b[1] - a[1]);
    let count = 0;
    for (let i = 0; i < entries.length; i++) {
        count += entries[i][1];
        if (count >= N) {
            return i + 1;
        }
    }
    return 0;
}

File: src/LCS02.WqXACV.1000226/solution.java
class Solution {
    public int halfQuestions(int[] questions) {
        int N = questions.length/2;
        Map<Integer,Long> map = Arrays.stream(questions).boxed().collect(Collectors.groupingBy(Function.identity(),Collectors.counting()));

        List<Long> list = map.values().stream().sorted(Comparator.reverseOrder()).toList();
        
        int result = 0;
        for(long count:list){
            result++;
            N -= (int)count;
            if(N<=0){
                break;
            }
        }
        return result;
    }
}